02-peer-protocol.md

```diff
@@ -787,9 +787,21 @@ is destined, is described in [BOLT #4](04-onion-routing.md).
 #### Requirements
 
 A sending node:
-  - MUST NOT offer `amount_msat` it cannot pay for in the
-remote commitment transaction at the current `feerate_per_kw` (see "Updating
-Fees") while maintaining its channel reserve.
+  - if it is _responsible_ for paying the Bitcoin fee:
+    - MUST NOT offer `amount_msat` if, after adding that HTLC to its commitment
+    transaction, it cannot pay the fee for either the local or remote commitment
+    transaction at the current `feerate_per_kw` while maintaining its channel
+    reserve (see "Updating Fees").
+    - SHOULD NOT offer `amount_msat` if, after adding that HTLC to its commitment
+    transaction, its remaining balance doesn't allow it to pay the fee for a
+    future additional non-dust HTLC at `N*feerate_per_kw` while maintaining its
+    channel reserve ("fee spike buffer"), where `N` is a parameter chosen by the
+    implementation (`N = 2` is recommended to ensure predictability).
```

Comment:
I am wondering now, does it make sense to include any formula at all in the spec? This `N*feerate` is just an idea. A different implementation may for example just use a fixed buffer regardless of what the current fee rate is.

Reply:
You're right, we can give more leeway here and just mention this as a suggestion. I think it's still worth having as a suggestion as it's the most natural way we came up with, but it's not a hard requirement.

Reply:
@joostjager  Kinda agree, but we should definitively mention `2*feerate_per_kw` as a suggestion to maintain some kind of predictability.

Reply:
Done in f5490f17d17ff49dc26ee459432b3c9db4fda8a9